RecoTauTag/RecoTau/plugins/PATTauDiscriminationByMVAIsolationRun2.cc

```diff
@@ -0,0 +1,306 @@
+    explicit PATTauDiscriminationByMVAIsolationRun2(const edm::ParameterSet& cfg)
+      : PATTauDiscriminationProducerBase(cfg),
+        moduleLabel_(cfg.getParameter<std::string>("@module_label")),
+	mvaReader_(0),
+	mvaInput_(0),
+	category_output_(0)
+    {
+       mvaName_ = cfg.getParameter<std::string>("mvaName");
+       loadMVAfromDB_ = cfg.exists("loadMVAfromDB") ? cfg.getParameter<bool>("loadMVAfromDB") : false;
+       if ( !loadMVAfromDB_ ) {
+         if(cfg.exists("inputFileName")){
+	   inputFileName_ = cfg.getParameter<edm::FileInPath>("inputFileName");
+	 }else throw cms::Exception("MVA input not defined") << "Requested to load tau MVA input from ROOT file but no file provided in cfg file";
+       }    
+       std::string mvaOpt_string = cfg.getParameter<std::string>("mvaOpt");
+       if      ( mvaOpt_string == "oldDMwoLT" ) mvaOpt_ = kOldDMwoLT;
+       else if ( mvaOpt_string == "oldDMwLT"  ) mvaOpt_ = kOldDMwLT;
+       else if ( mvaOpt_string == "newDMwoLT" ) mvaOpt_ = kNewDMwoLT;
+       else if ( mvaOpt_string == "newDMwLT"  ) mvaOpt_ = kNewDMwLT;
+       else if ( mvaOpt_string == "DBoldDMwLT"  ) mvaOpt_ = kDBoldDMwLT;
+       else if ( mvaOpt_string == "DBnewDMwLT"  ) mvaOpt_ = kDBnewDMwLT;
+       else if ( mvaOpt_string == "PWoldDMwLT"  ) mvaOpt_ = kPWoldDMwLT;
+       else if ( mvaOpt_string == "PWnewDMwLT"  ) mvaOpt_ = kPWnewDMwLT;
+       else throw cms::Exception("PATTauDiscriminationByMVAIsolationRun2")
+         << " Invalid Configuration Parameter 'mvaOpt' = " << mvaOpt_string << " !!\n";
+		    
+       if      ( mvaOpt_ == kOldDMwoLT || mvaOpt_ == kNewDMwoLT ) mvaInput_ = new float[6];
+       else if ( mvaOpt_ == kOldDMwLT  || mvaOpt_ == kNewDMwLT  ) mvaInput_ = new float[12];
+       else if ( mvaOpt_ == kDBoldDMwLT || mvaOpt_ == kDBnewDMwLT ||
+                 mvaOpt_ == kPWoldDMwLT || mvaOpt_ == kPWnewDMwLT) mvaInput_ = new float[23];
+       else assert(0);
+
+       chargedIsoPtSums_ = cfg.getParameter<std::string>("srcChargedIsoPtSum");
+       neutralIsoPtSums_ = cfg.getParameter<std::string>("srcNeutralIsoPtSum");
+       puCorrPtSums_ = cfg.getParameter<std::string>("srcPUcorrPtSum");
+       photonPtSumOutsideSignalCone_ = cfg.getParameter<std::string>("srcPhotonPtSumOutsideSignalCone");
+       footprintCorrection_ = cfg.getParameter<std::string>("srcFootprintCorrection");
+		  
+       verbosity_ = ( cfg.exists("verbosity") ) ?
+         cfg.getParameter<int>("verbosity") : 0;
+
+       produces<pat::PATTauDiscriminator>("category");
+    }  
+		
+    void beginEvent(const edm::Event&, const edm::EventSetup&);
+		
+    double discriminate(const TauRef&) const;
+		
+    void endEvent(edm::Event&);
+		
+    ~PATTauDiscriminationByMVAIsolationRun2()
+    {
+      if(!loadMVAfromDB_) delete mvaReader_;
+      delete[] mvaInput_;
+      for ( std::vector<TFile*>::iterator it = inputFilesToDelete_.begin();
+	    it != inputFilesToDelete_.end(); ++it ) {
+        delete (*it);
+      }
+    }
```

Comment:
I can have a look and try to improve on it in following code improvement commits.